hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/StripedDataStreamer.java

```diff
@@ -111,6 +111,7 @@ protected LocatedBlock nextBlockOutputStream() throws IOException {
       final DatanodeInfo badNode = nodes[getErrorState().getBadNodeIndex()];
       LOG.warn("Excluding datanode " + badNode);
       excludedNodes.put(badNode, badNode);
+      setPipeline(null, null, null);
```

Comment:
Why is this required, we are anyway throwing, is it because we called ``setPipeline(lb);`` above? why don't we call it just before the ``return`` statement? Originally we are returning this `lb` & then setting the pipeline post that, so we can do this just before ``return lb``?

Reply:
The pipeline which is set here is used in setupPipelineForAppendOrRecovery which will now be used in this flow. If that also fails, we set it to null to clear the existing state.

Reply:
That is in {{DataStreamer.java}} right? Is it being called for StripedDataStreamer.java as well?

Reply:
Updated